### ISSUE TITLE
signal-lab: wandb integration + full pipeline

### ISSUE THREAD

**Issue by andriiSad** [andriiSad/ai-trader#35]
## Parent

PRD: #29

## What to Build

Integrate wandb for full experiment tracking across all 3 models. Build the pipeline orchestrator that runs the complete model progression.

**New files:**
- `src/wandb_logger.py` — wandb logging wrapper
- `src/pipeline.py` — full pipeline orchestrator
- `requirements.txt` update: add wandb

**wandb logger (`src/wandb_logger.py`):**
- `init_run(project, model_name, fold_idx, config)` — initialize wandb run
- `log_metrics(metrics, step)` — log accuracy, F1, loss, confusion matrix
- `log_artifact(name, type, path)` — save model checkpoint, feature importance
- `log_feature_importance(model, feature_names)` — log feature importance bar chart
- `finish_run()` — close wandb run
- Project name: `signal-lab`
- Run names: `lr-fold-1`, `lgbm-fold-1`, `lstm-fold-1`

**Pipeline (`src/pipeline.py`):**
- `run_pipeline(config, models=None)` — orchestrates full pipeline:
  1. Load data
  2. Generate features + labels
  3. For each model (lr, lgbm, lstm):
     - For each walk-forward fold:
       - Fit scaler on train
       - Transform train + test
       - Train model
       - Compute metrics
       - Log to wandb
  4. Print summary table comparing all models

**CLI update:**
- `signal_lab.py run` — full pipeline (download → features → train all models)
- `signal_lab.py run --model lr` — single model
- `signal_lab.py run --model lgbm` — single model
- `signal_lab.py run --model lstm` — single model

**Requirements update:**
- Add `wandb>=0.17` to requirements.txt

**Tests:**
- test_pipeline.py: test pipeline runs with mocked wandb, test model selection

## Acceptance Criteria

- [ ] `src/wandb_logger.py` implements init, log, artifact, finish functions
- [ ] wandb project name is `signal-lab`
- [ ] Run names follow pattern: `{model}-fold-{n}`
- [ ] Metrics logged per fold: accuracy, F1, loss, confusion matrix
- [ ] Artifacts logged: model checkpoints, feature importance
- [ ] `src/pipeline.py` orchestrates all 3 models across all folds
- [ ] `signal_lab.py run` executes full pipeline
- [ ] `signal_lab.py run --model {lr|lgbm|lstm}` runs single model
- [ ] Summary table printed at end comparing all models
- [ ] `requirements.txt` includes wandb
- [ ] All tests pass
- [ ] Root `make lint` passes

## Blocked by

- #32 (walk-forward + logistic regression)
- #33 (LightGBM model)
- #34 (PyTorch LSTM model)

## Shared Interfaces

```python
# From all model tasks (#32, #33, #34)
def train_logistic(X_train, y_train, X_test, y_test) -> dict
def train_lightgbm(X_train, y_train, X_test, y_test) -> dict
def train_lstm(X_train, y_train, X_test, y_test) -> dict

# From src/walk_forward.py (provided by #32)
def split(df: pd.DataFrame, train_months: int = 6, test_months: int = 2) -> Iterator[tuple[pd.DataFrame, pd.DataFrame]]

# From src/metrics.py (provided by #32)
def compute_metrics(y_true, y_pred, y_prob) -> dict

# This task provides:
def init_run(project, model_name, fold_idx, config) -> None
def log_metrics(metrics, step=None) -> None
def log_artifact(name, artifact_type, path) -> None
def finish_run() -> None
def run_pipeline(config, models=None) -> dict
```
